### ISSUE TITLE
Release MADD model and SAM-FB dataset on Hugging Face

### ISSUE THREAD

**Issue by NielsRogge** [KaKituken/affordance-aware-any#1]
Hello @KaKituken 🤗 

I'm Niels and work as part of the open-source team at Hugging Face. I discovered your work through Hugging Face's daily papers as yours got featured: https://huggingface.co/papers/2412.14462.
The paper page lets people discuss about your paper and lets them find artifacts about it (your models for instance), you can also claim the paper as yours which will show up on your public profile at HF.

Would you like to host the model you've pre-trained on https://huggingface.co/models and datasets you've released on https://huggingface.co/datasets?
Hosting on Hugging Face will give you more visibility/enable better discoverability. We can add tags in the model and dataset cards so that people find them easier, link them to the paper page, etc.

If you're down, leaving a guide [here](https://huggingface.co/docs/hub/models-uploading) for the models and here [https://huggingface.co/docs/datasets/loading](https://huggingface.co/docs/datasets/loading) for the datasets.

If it's a custom PyTorch model, you can use the [PyTorchModelHubMixin](https://huggingface.co/docs/huggingface_hub/package_reference/mixins#huggingface_hub.PyTorchModelHubMixin) class which adds `from_pretrained` and `push_to_hub` to the model which lets you to upload the model and people to download and use models right away.
If you do not want this and directly want to upload model through UI or however you want, people can also use [hf_hub_download](https://huggingface.co/docs/huggingface_hub/en/guides/download#download-a-single-file).

Besides that, there's the [dataset viewer](https://huggingface.co/docs/hub/en/datasets-viewer) which allows people to quickly explore the first few rows of the data in the browser.

After uploaded, we can also link the models and datasets to the paper page (read [here](https://huggingface.co/docs/hub/en/model-cards#linking-a-paper)) so people can discover your model.

You can also build a demo for your model on [Spaces](https://huggingface.co/spaces), we can provide you an A100 grant.

Let me know if you're interested/need any guidance.

Kind regards,

Niels

**Comment by KaKituken** [comment]
font{
        line-height: 1.6;
    }
    ul,ol{
        padding-left: 20px;
        list-style-position: inside;
    }


    
    Hi Niels,Thank you for your email.I am trying to upload the dataset to hugging face. However I have 1k sub-folders of 3GB, which makes the total size about 3T. I am considering to maybe upload a part of them so that they can fit the storage limitation. May I ask what are your suggestions?Best,Jixuan
    
        
            
        
    
    
      
    
        
            
    
        
            
                    
                        
                    
                    
                        hejixuan2000
                    
            
                
                    
                            ***@***.***
                    
                
        
    

        
    
 
    

  
    ---- Replied Message ----
  
  
    
      
        
         From 
        
        
            ***@***.***>
            
        
      
      
        
         Date 
        
        
        12/20/2024 21:12
        
      
      
        
         To 
        
        
         
          
            ***@***.***>
            
          
        
      
      
        
         Cc 
        
        
          
            Jixuan ***@***.***>
            ,
          
          
            ***@***.***>
            
          
        
      
      
        
         Subject 
        
        
              [KaKituken/affordance-aware-any] Release MADD model and SAM-FB dataset on Hugging Face (Issue #1)
        
      
    
  
  
Hello @KaKituken 🤗
I'm Niels and work as part of the open-source team at Hugging Face. I discovered your work through Hugging Face's daily papers as yours got featured: https://huggingface.co/papers/2412.14462.
The paper page lets people discuss about your paper and lets them find artifacts about it (your models for instance), you can also claim the paper as yours which will show up on your public profile at HF.
Would you like to host the model you've pre-trained on https://huggingface.co/models and datasets you've released on https://huggingface.co/datasets?
Hosting on Hugging Face will give you more visibility/enable better discoverability. We can add tags in the model and dataset cards so that people find them easier, link them to the paper page, etc.
If you're down, leaving a guide here for the models and here https://huggingface.co/docs/datasets/loading for the datasets.
If it's a custom PyTorch model, you can use the PyTorchModelHubMixin class which adds from_pretrained and push_to_hub to the model which lets you to upload the model and people to download and use models right away.
If you do not want this and directly want to upload model through UI or however you want, people can also use hf_hub_download.
Besides that, there's the dataset viewer which allows people to quickly explore the first few rows of the data in the browser.
After uploaded, we can also link the models and datasets to the paper page (read here) so people can discover your model.
You can also build a demo for your model on Spaces, we can provide you an A100 grant.
Let me know if you're interested/need any guidance.
Kind regards,
Niels

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

**Comment by lhoestq** [comment]
Hi ! Quentin from HF here 🤗

We can provide you with a storage grant if the dataset is in a standard format like WebDataset. This enables easy loading in most training frameworks, and also to enable the Dataset Viewer on HF :)

Would it be possible for you to format the dataset as a WebDataset ?

WebDataset is based on TAR, similar to your current format, but has one metadata file per image. See how to structure the TAR archives for WebDataset here: https://huggingface.co/docs/hub/en/datasets-webdataset
